src/nsrt_learning/side_predicate_learning.py

```diff
@@ -487,8 +576,8 @@ def _try_specializing_pnad(
         # Get an arbitrary grounding of the PNAD's operator whose
         # preconditions hold in segment.init_atoms and whose add
         # effects are a subset of necessary_add_effects.
-        ground_op = self._get_partially_satisfying_grounding(
-            necessary_add_effects, pnad, segment)
+        ground_op = self._find_unification(necessary_add_effects, pnad,
+                                           segment, True)
```

Comment:
generally preferred to use keyword argument when passing in a constant thing, e.g. `..., find_partial_grounding=True)`